app/src/test/java/com/example/nvdovin/weatherapp/presentation/details/core/DetailsPresenterTest.java

```diff
@@ -0,0 +1,97 @@
+package com.example.nvdovin.weatherapp.presentation.details.core;
+
+import android.content.Context;
+
+import com.example.nvdovin.weatherapp.data.model.City;
+import com.example.nvdovin.weatherapp.data.model.WeatherData;
+import com.example.nvdovin.weatherapp.domain.model.DailyForecast;
+import com.example.nvdovin.weatherapp.domain.service.CityService;
+import com.example.nvdovin.weatherapp.domain.service.WeatherDataService;
+import com.example.nvdovin.weatherapp.domain.utils.mapper.DataMapper;
+import com.example.nvdovin.weatherapp.domain.utils.navigator.Navigator;
+import com.example.nvdovin.weatherapp.domain.utils.sharedpreferences.SharedPrefs;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.List;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DetailsPresenterTest {
+
+    private final static Long MOCK_LONG_VALUE = (long) Math.random();
+    @Mock
+    CityService cityService;
+    @Mock
+    WeatherDataService weatherDataService;
+    @Mock
+    DataMapper dataMapper;
+    @Mock
+    DetailView detailView;
+    @Mock
+    Context context;
+    //-------
+    @Mock
+    City city;
+    @Mock
+    List<DailyForecast> dailyForecastList;
+    @Mock
+    WeatherData weatherData;
+    @Mock
+    Navigator.Builder builder;
+    @Mock
+    SharedPrefs sharedPreferences;
+    private DetailsPresenter detailsPresenter;
+
+    @Before
+    public void setUp() throws Exception {
+        detailsPresenter = new DetailsPresenter(cityService,
+                weatherDataService,
+                dataMapper,
+                detailView,
+                builder,
+                sharedPreferences);
+
+        when(cityService.getCityById(anyLong())).thenReturn(city);
```

Comment:
I don't really like to have all this when...thenReturn... in a setUp method.
In this case, in setUp() only presenter should be initialised, and retrieve this common when...then... block in a method that will be called by test methods where needed.